vtpm-snp/src/hcl.rs

```diff
@@ -101,27 +116,34 @@ fn buf_to_hcl_data(bytes: &[u8]) -> Result<(HclAttestationReport, &[u8]), Box<dy
     Ok((hcl_report, var_data))
 }
 
+#[derive(Error, Debug)]
+pub enum ParseError {
+    #[error("bincode error")]
+    Bincode(#[from] Box<bincode::ErrorKind>),
+    #[error("json parse error")]
+    Report(#[from] serde_json::Error),
+}
+
 impl TryFrom<&[u8]> for HclReportWithRuntimeData {
-    type Error = Box<dyn error::Error>;
+    type Error = ParseError;
 
     fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
         let (hcl_report, var_data) = buf_to_hcl_data(bytes)?;
         let runtime_data: ReportData = serde_json::from_slice(var_data)?;
+        let runtime_data_hash = sha256(var_data);
```

Comment:
I don't think this is helpful. The verifier can just as well look into the snp report and extract the hash from there.

If the verifier is to trust the HCL AK key, then they need to see that it is contained in the snp report. For that they need to do the hashing themselves.

Reply:
mhh, in fact we don't need to have a parsed report in the evidence, we don't do anything w/ it on the attester's side. so we can send a raw report and have the verifier parse it and compute hashes.